ext/opentelemetry-ext-elasticsearch/src/opentelemetry/ext/elasticsearch/__init__.py

```diff
@@ -0,0 +1,155 @@
+    es.get(index='my-index', doc_type='my-type', id=1)
+
+API
+---
+
+# TODO: mention config params
+"""
+
+import functools
+import types
+from logging import getLogger
+from os import environ
+
+import elasticsearch
+import elasticsearch.exceptions
+from wrapt import ObjectProxy
+from wrapt import wrap_function_wrapper as _wrap
+
+from opentelemetry import context, propagators, trace
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.ext.elasticsearch.version import __version__
+from opentelemetry.trace import SpanKind, get_tracer
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+logger = getLogger(__name__)
+
+
+# Values to add as tags from the actual
+# payload returned by Elasticsearch, if any.
+_attributes_from_result = [
+    "found",
+    "timed_out",
+    "took",
+]
+
+_default_op_name = "request"
+
+
+class ElasticsearchInstrumentor(BaseInstrumentor):
+    """An instrumentor for requests
+    See `BaseInstrumentor`
+    """
+
+    def __init__(self, span_name_prefix=None):
+        if not span_name_prefix:
+            span_name_prefix = environ.get(
+                "OPENTELEMETRY_PYTHON_ELASTICSEARCH_NAME_PREFIX",
+                "Elasticsearch",
+            )
+        self.span_name_prefix = span_name_prefix.strip()
+        super().__init__()
+
+    def _instrument(self, **kwargs):
+        """
+        Instruments elasticsarch module
+        """
+        tracer_provider = kwargs.get("tracer_provider")
+        tracer = get_tracer(__name__, __version__, tracer_provider)
+        _wrap(
+            elasticsearch,
+            "Transport.perform_request",
+            _wrap_perform_request(tracer, self.span_name_prefix),
+        )
+
+    def _uninstrument(self, **kwargs):
+        _unwrap(elasticsearch.Transport, "perform_request")
+
+
+def _wrap_perform_request(tracer, span_name_prefix):
+    def wrapper(wrapped, _, args, kwargs):
+        method = url = None
+        try:
+            method, url = args
+        except IndexError:
+            logger.warning(
+                "expected perform_request to receive two positional arguments. "
+                "Got %d",
+                len(args),
+            )
+
+        op_name = span_name_prefix + (url or method or _default_op_name)
+        params = kwargs.get("params", {})
+        body = kwargs.get("body", None)
+
+        attributes = {
+            "component": "elasticsearch-py",
+            "db.type": "elasticsearch",
+        }
+
+        if url:
+            attributes["elasticsearch.url"] = url
+        if method:
+            attributes["elasticsearch.method"] = method
+        if body:
+            attributes["db.statement"] = str(body)
+        if params:
+            attributes["elasticsearch.params"] = params
+
+        with tracer.start_as_current_span(
+            op_name, kind=SpanKind.CLIENT, attributes=attributes
```

Comment:
Is the ability to add a prefix to the span name unique to elasticsearch? I don't see it for other integrations.

Reply:
I migrated this from opentracing which had this. I don't remember other opentracing libs having a similar feature but I may be wrong. Will confirm.

Reply:
This seems to be specific to elasticsearch in opentracing. I'm not sure why but I'd like to keep to make the upgrade path for users moving from OT to Otel as smooth as possible.